src/dynarec/la64/dynarec_la64_66.c

```diff
@@ -156,8 +157,8 @@ uintptr_t dynarec64_66(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
             if (MODREG) {
                 ed = TO_LA64((nextop & 7) + (rex.b << 3));
                 if (ed != gd) {
-                    BSTRINS_W(ed, gd, 15, 0);
-                    ZEROUP(ed);
+                    BSTRINS_D(ed, gd, 15, 0);
+                    if (rex.is32bits) ZEROUP(ed);
```

Comment:
Yes, `BSTRINS.W` does sign-extension on high 32bits.

Reply:
That's why the `BFIz` trick on ARM64 does not work here, need 2 instructions for the 32-bit situation.

Reply:
> That's why the `BFIz` trick on ARM64 does not work here, need 2 instructions for the 32-bit situation.

In that case, this will not work on 64bits if the upper 32bits is wipped!

Reply:
That why I changed it to `BSTRINS.D` ;)

And you know what, this suddenly fixed Terraria on LA64... How lucky is me, mono games are hard to debug..

Reply:
But then, if BSTRINS.D doesn't change the upper 32bits (wich is what we want), the ZEROUP is not needed.

Reply:
This is ported from ARM64, from

```
BFIz(rd, rs, 0, 16);
```
to
```
BSTRINS_D(rd, rs, 15, 0);
if (rex.is32bits) ZEROUP(rd);
```

Reply:
But again, the ZEROUP should not be needed, BSTRINS_D should not  zero extend 32bits->64bits